### ISSUE TITLE
Results difference between geometries_from_polygon() and geometries_from_place()

### ISSUE THREAD

**Issue by RolandDeschain19** [gboeing/osmnx#958]
**Do not delete the following template. Fill it out completely.**

  - [x] I have read the OSMnx [contributing guidelines](../blob/main/CONTRIBUTING.md)
  - [x] I have read the OSMnx [documentation](https://osmnx.readthedocs.io/) and [usage examples](https://github.com/gboeing/osmnx-examples)
  - [x] I have searched the open/closed issues and [StackOverflow](https://stackoverflow.com/search?q=osmnx) for problems similar to mine

**Problem description**
  - What did you do?
  - I used the **geometries_from_polygon()** function to retrieve all the nodes with tag "barrier:toll_booth" inside italian regions, using the shapefiles found at the official italian website https://www.istat.it/it/archivio/222527, "Confini Amministrativi 2023.zip". I then used the **geometries_from_place()** function to retrieve the same data, specifying the region (i.e. "Emilia Romagna, Italy").
  - What did you expect to happen?
  - I expected the results to be equals using the two functions on the same region.
  - And what actually happened instead?
  - It seems that using the **geometries_from_polygon** function, the nodes obtained are outdated. For example, for the Emilia Romagna region I obtain node 1690224666 in Bologna, which has been deleted 6 days ago. Using the **geometries_from_place** function, I obtain the correct node, 10904918661, created 6 days ago in the same position.

**Environment information**
  - What operating system are you using?
  - Linux version 3.10.0-1160.49.1.el7.x86_64 (mockbuild@x86-vm-43.build.eng.bos.redhat.com) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC)
  - What Python version are you using?
  - Python 3.6.8
  - What OSMnx version are you using?
  - 1.2.2
  - Provide a complete list of your environment's packages and their versions (for example, run `conda list` or `pip list` then paste the output between the two "details" tags below)

<details>
Package                      Version
---------------------------- -------------------
absl-py                      1.0.0
adjustText                   0.8
argon2-cffi                  20.1.0
astunparse                   1.6.3
async-generator              1.10
atalanta                     0.1.0
atpublic                     2.3
attrs                        20.3.0
backcall                     0.2.0
bcrypt                       3.2.0
beautifulsoup4               4.9.3
bitarray                     1.6.3
bleach                       3.3.0
branca                       0.6.0
cached-property              1.5.2
cachetools                   5.0.0
cdsapi                       0.5.1
certifi                      2020.12.5
cffi                         1.14.4
cftime                       1.5.1
chardet                      4.0.0
charset-normalizer           3.1.0
click                        7.1.2
click-plugins                1.1.1
cligj                        0.7.1
cloudpickle                  1.6.0
clusteval                    2.1.5
clustimage                   1.5.14
cmdstanpy                    0.9.68
colourmap                    1.1.10
contourpy                    1.0.7
convertdate                  2.3.2
cryptography                 36.0.1
cycler                       0.10.0
Cython                       0.29.21
dask                         2021.6.2
dataclasses                  0.6
decorator                    4.4.2
defusedxml                   0.6.0
distfit                      1.6.10
docopt                       0.6.2
ecate                        1.0.3
entrypoints                  0.3
ephem                        3.7.7.1
et-xmlfile                   1.0.1
fabric                       2.6.0
Fabric3                      1.14.post1
Fiona                        1.8.18
flatbuffers                  2.0
folium                       0.14.0
fonttools                    4.39.2
fsspec                       2021.6.1
gast                         0.5.3
geographiclib                2.0
geopandas                    0.12.2
geopy                        2.3.0
google-auth                  2.6.6
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
graphviz                     0.8.4
grpcio                       1.44.0
gssapi                       1.7.3
h3                           3.7.3
h5py                         3.6.0
haversine                    2.7.0
hdfs                         2.6.0
hijri-converter              2.1.1
holidays                     0.11.1
ibis-framework               2.1.1
idna                         2.10
ImageHash                    4.3.1
imageio                      2.27.0
importlib-metadata           4.11.3
importlib-resources          5.12.0
impyla                       0.17.0
install                      1.3.4
invoke                       1.6.0
ipykernel                    5.4.3
ipyleaflet                   0.13.6
ipython                      7.20.0
ipython-genutils             0.2.0
ipywidgets                   7.6.3
ismember                     1.0.2
jedi                         0.18.0
Jinja2                       2.11.3
joblib                       1.1.0
jsonschema                   3.2.0
jupyter                      1.0.0
jupyter-client               6.1.11
jupyter-console              6.2.0
jupyter-core                 4.7.1
jupyterlab-pygments          0.1.2
jupyterlab-widgets           1.0.0
kaleido                      0.2.1
kepler                       0.0.1
keplergl                     0.3.2
keras                        2.8.0
Keras-Preprocessing          1.1.2
kerberos                     1.3.1
kiwisolver                   1.3.1
korean-lunar-calendar        0.2.1
kPOD                         0.18
krb5                         0.3.0
krbcontext                   0.10
lazy_loader                  0.2
libclang                     14.0.1
llvmlite                     0.39.1
locket                       0.2.1
LunarCalendar                0.0.9
lxml                         4.6.3
Markdown                     3.3.6
MarkupSafe                   1.1.1
matplotlib                   3.7.1
mistune                      0.8.4
multipledispatch             0.6.0
munch                        2.5.0
mxnet-mkl                    1.6.0
mysql                        0.0.3
mysql-connector-python       8.0.31
mysqlclient                  2.1.1
nbclient                     0.5.2
nbconvert                    6.0.7
nbformat                     5.1.2
nest-asyncio                 1.5.1
netCDF4                      1.5.7
networkx                     3.0
notebook                     6.2.0
numba                        0.56.4
numpy                        1.23.1
oauthlib                     3.2.0
opencv-python                4.7.0.72
openpyxl                     3.0.7
opt-einsum                   3.3.0
osmnx                        1.2.2
overpy                       0.6
packaging                    21.3
pandas                       2.0.1
pandas-datareader            0.10.0
pandocfilters                1.4.3
paramiko                     2.9.2
parso                        0.8.1
parsy                        1.4.0
partd                        1.2.0
pathlib2                     2.3.7.post1
patsy                        0.5.2
pca                          1.9.2
pexpect                      4.8.0
pickleshare                  0.7.5
Pillow                       9.5.0
pip                          23.0.1
plotly                       4.14.3
ply                          3.11
progress                     1.5
prometheus-client            0.9.0
prompt-toolkit               3.0.16
protobuf                     3.20.1
psutil                       5.9.2
ptyprocess                   0.7.0
pure-sasl                    0.6.2
pyasn1                       0.4.8
pyasn1-modules               0.2.8
pycparser                    2.20
Pygments                     2.7.4
pygrib                       2.1.4
PyMeeus                      0.5.11
PyNaCl                       1.5.0
pynndescent                  0.5.8
pyparsing                    2.4.7
pypickle                     1.1.0
pyproj                       3.4.1
pyrsistent                   0.17.3
pyshp                        2.1.3
pyspnego                     0.5.0
python-dateutil              2.8.2
python-Levenshtein           0.12.2
pytz                         2021.1
PyWavelets                   1.4.1
PyYAML                       5.4.1
pyzmq                        22.0.2
qtconsole                    5.0.2
QtPy                         1.9.0
regex                        2021.11.10
requests                     2.28.2
requests-kerberos            0.14.0
requests-oauthlib            1.3.1
retrying                     1.3.3
rsa                          4.8
Rtree                        1.0.1
sasl                         0.2.1
scatterd                     1.2.5
scikit-image                 0.20.0
scikit-learn                 1.0.2
scikit-learn-extra           0.2.0
scipy                        1.9.1
seaborn                      0.12.0
selenium                     3.141.0
Send2Trash                   1.5.0
setuptools                   49.2.1
setuptools-git               1.2
Shapely                      1.8.5.post1
six                          1.15.0
sklearn                      0.0
soupsieve                    2.2.1
statsmodels                  0.13.2
tensorboard                  2.8.0
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorflow                   2.8.0
tensorflow-io-gcs-filesystem 0.25.0
termcolor                    1.1.0
terminado                    0.9.2
testpath                     0.4.4
tf-estimator-nightly         2.8.0.dev2021122109
threadpoolctl                3.1.0
thrift                       0.11.0
thrift-sasl                  0.4.3
thriftpy                     0.3.9
thriftpy2                    0.4.0
tifffile                     2023.3.21
toolz                        0.11.1
tornado                      6.1
tqdm                         4.59.0
traitlets                    5.0.5
traittypes                   0.2.1
typing_extensions            4.2.0
tzdata                       2023.3
ujson                        4.0.2
umap-learn                   0.5.3
urllib3                      1.26.3
wcwidth                      0.2.5
webencodings                 0.5.1
Werkzeug                     2.1.1
wget                         3.2
wheel                        0.37.1
widgetsnbextension           3.5.1
wrapt                        1.14.0
yellowbrick                  1.5
zipp                         3.8.0
</details>

**Provide a complete minimal reproducible example**

Your example code snippet here must be *minimal* so it doesn't contain extraneous code or data unrelated to your specific problem and it must be *complete* so we can independently run it from top to bottom by copying/pasting it into a Python interpreter. That means all imports and all variables must be defined. If you're unsure how to create a good reproducible example, read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports). Do not post a screenshot of your code or error message: provide it as text.

```python

import osmnx as ox

tags = {'barrier': 'toll_booth'}
place_name = "Emilia Romagna, Italy"

toll_place = ox.geometries_from_place(place_name, tags)
```

**Comment by gboeing** [comment]
If you can provide precise code to exactly reproduce the bug, I can try to take another look. But I'm not able to see it as of now.

**Comment by RolandDeschain19** [comment]
@gboeing thank you for the fast reply!

I am sorry for not having provided the code in full, but the bug template explicitly said to include a full working code from copy-paste, without external elements.

If you download the shapefiles zip I linked in the original post, you can then use this code to try and replicate my results.

```python

import osmnx as ox
import geopandas as gpd
import pandas as pd

italy = gpd.read_file('Limiti01012023/Reg01012023/Reg01012023_WGS84.shp')
italy_polygon = italy.geometry
italy_prj = italy.to_crs(4326)

tags = {'barrier': 'toll_booth'}
toll_poly = ox.geometries_from_polygon(italy_prj.geometry[7], tags)
```

Meanwhile, I will upgrade osmnx and try with the latest release.

**Comment by gboeing** [comment]
I'm still getting the same results I did earlier... no bug seen. Code:

```python
import osmnx as ox
import geopandas as gpd
import pandas as pd
ox.settings.log_console = True
tags = {'barrier': 'toll_booth'}
place_name = "Emilia Romagna, Italy"

# from place
tp = ox.geometries_from_place(place_name, tags)
print(tp.loc[('node', 10904918661)]) # success
print(tp.loc[('node', 1690224666)])  # error

# from polygon
italy = gpd.read_file('Limiti01012023/Reg01012023/Reg01012023_WGS84.shp')
italy_polygon = italy.geometry
italy_prj = italy.to_crs(4326)
tp = ox.geometries_from_polygon(italy_prj.geometry[7], tags)
print(tp.loc[('node', 10904918661)]) # success
print(tp.loc[('node', 1690224666)])  # error
```

**Comment by AtelierLibre** [comment]
Apologies if wide of the mark but there is no chance this is a caching issue is it? Two slightly different queries run a few days apart with different data in their caches?

**Comment by gboeing** [comment]
@RolandDeschain19 you can test @AtelierLibre's hypothesis by setting `ox.settings.use_cache = False`.

**Comment by RolandDeschain19** [comment]
@gboeing @AtelierLibre it seems that the cache was indeed the issue! By using `ox.settings.use_cache = False` now I obtain the same result using the two functions.
Thank you a lot for the help and your time!
